tests/Microsoft.Identity.Test.Integration.netfx/HeadlessTests/ClientCredentialsTests.WithRegion.cs

```diff
@@ -46,25 +46,18 @@ public void TestInitialize()
             }
         }
 
-        [TestCleanup]
-        public void TestCleanup()
-        {
-            Environment.SetEnvironmentVariable(TestConstants.RegionName, null);
-        }
-
         [TestMethod]
         public async Task AcquireTokenToRegionalEndpointAsync()
         {
             // Arrange
             var factory = new HttpSnifferClientFactory();
             var settings = ConfidentialAppSettings.GetSettings(Cloud.Public);
-            _confidentialClientApplication = BuildCCA(settings, factory);
+            _confidentialClientApplication = BuildCCA(settings, factory, false, TestConstants.Region);
```

Comment:
I am not 100% sure if this test is right with the removal of the env variable. I assumed that if I set the same region with the WithAzureRegion() api that the test would attempt discovery via IMDS but the call fails with {"A socket operation was attempted to an unreachable network."}. Is IMDS supposed to be reachable in this integration test with "centralis"?

@bgavrilMS

Reply:
IMDS is an HTTP call to smth like `http://192.168.0.1/imds/metadata`. This url is not available on your local machine, and so it fails. 

However, it might work on the Azure DevOps VM, since they use actual VMs.

Reply:
Seems like it is working on the cloud build but not locally. The is resulting in  test issue however since the behavior of the test is different when run locally, resulting in assertions to fail. Will need to resolve

Reply:
How about 

```csharp
if (Environment.GetVariable("TF_BUILD") == null)
{
   Assert.Inconclusive("This test needs to run on Azure DevOps hosted agent, to be located on an Azure VM");
}
```

https://docs.microsoft.com/en-us/azure/devops/pipelines/build/variables?view=azure-devops&tabs=yaml